naming-things.md

```diff
@@ -0,0 +1,200 @@
+# Naming Conventions
+
+Here are some go-to conventions for the names of things.
+
+
+## Variables
+
+
+### Looping over dictionaries
+
+You **should** use the names `key` and `value`
+
+```python
+for key, value in thing.items():
+    ...
+```
+
+
+### Looping over iterables
+
+
+When writing generic loops, you **should** use the name ``value``.
+
+```python
+for value in things_to_process:
+    ...
+```
+
+In cases where the term ``value`` is not available, you **should** use the term
+``item``
+
+```python
+for key, value in thing.items():
+    for item in value:
+        ...
+```
+
+
+### Confusing Plurals
+
+For short names, the simple plural form is preferred.
+
+```python
+for user in users:
+    ...
+```
+
+
+For longer names, avoid using plural names that only differ by one letter.
+
+```python
+for lockfile_path in lockfile_paths:
+    ...
+```
+
+When using longer names, consider adding a prefix or suffix to help
+differentiate the variable names.
+
+```python
+for lockfile_path in lockfile_paths_to_process:
+    ...
+
+for user in all_users:
+    ...
```

Comment:
I'd like to keep it simply because it uses a prefix which while maybe less preferred is a pattern I've used in plenty of places...  and I'm lazy.

Reply:
I think I wasn't clear. The earlier example of `for user in users:` is declared as the preferred. So it's especially strange to have `users` be the example here. (I have no problem with the prefix `all_`). So I guess the idea is to use something longer, like... `doorbell`. Not sure why that was the first thing that came to mind X)

Reply:
ok, I didn't realized I had stated `for user in users` was preferred.  I updated that part to indicate that it *can* be ok but that it can still be visually confusing which I think removes the contradiction you were having issue with.